ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -269,33 +288,35 @@ class JITModBuilder {
         // function calls so we know when we left the recusion.
         OutlineCallDepth += 1;
         InlinedCalls.push_back(make_tuple(CurInstrIdx, CI));
-        return;
       }
       // If this is a recursive call that has been inlined, or if the callee
       // has the "yk_outline" annotation, remove the inlined code and turn it
       // into a normal (outlined) call.
-      if ((isRecursiveCall(CF) || CF->hasFnAttribute(YK_OUTLINE_FNATTR))) {
+      else if ((isRecursiveCall(CF) || CF->hasFnAttribute(YK_OUTLINE_FNATTR))) {
```

Comment:
The inner brackets are pointless as we have `((... || ...))` which is the same as `(... || ...)`.

Reply:
You are correct! Must be one of @vext01 hidden tests that you missed in a previous PR. ;-)

Reply:
`((...) || (...))` is idiomatic C++.

Reply:
@vext01 That isn't what's happening here though.

Reply:
Indeed.

Reply:
Fixed in 614f01f4d

Reply:
Lukas has gone with `(... || ...)`. I'm OK with it if you are Laurie.

Reply:
I don't follow. I thought this was just about there being an extra pair of outer brackets?